db/migrate/20170905201043_create_pages.rb

```diff
@@ -0,0 +1,10 @@
+class CreatePages < ActiveRecord::Migration[5.1]
+  def change
+    create_table :pages do |t|
+      t.string :key, unique: true, null: false
```

Comment:
I think the index must be specified like: 
```
      t.string :key, index: { unique: true }, null: false
```
in order to generate a unique index.